nengo/builder/connection.py

```diff
@@ -219,8 +219,11 @@ def get_prepost_signal(is_pre):
             targets.append(r.modifies)
 
         if 'encoders' in targets:
-            model.sig[conn.post_obj]['encoders'].readonly = False
-            model.add_op(PreserveValue(model.sig[conn.post_obj]['encoders']))
+            encoder_sig = model.sig[conn.post_obj]['encoders']
+            if not any(isinstance(op, PreserveValue) and op.dst is encoder_sig
+                       for op in model.operators):
```

Comment:
Hmm... it's probably not that expensive, but it feels like we shouldn't have to look through all the ops here. Feels like we should have a data structure keeping track of ops like we do for signals in `model.sig`... but for the time being, we do a quick hack of something like a `model.preservevalues` dict that is keyed by the encoder signal which we look at here. Does that make sense or is it overkill for how long this is likely to take? We could also defer that to a later time; if so, make a quick issue so we don't forget and then this looks good to me :)